utils/s2n_safety.h

```diff
@@ -93,6 +93,26 @@
  */
 #define ENSURE_NONNULL( x )                          ENSURE((x) != NULL, S2N_ERR_NULL)
 
+/**
+ * Ensures the `condition` is `true`, otherwise the function will `BAIL` with a `S2N_ERR_SAFETY` error
+ */
+#    define PRECONDITION( condition )                 ENSURE((condition), S2N_ERR_PRECONDITION_VIOLATION)
```

Comment:
looks like you have the indentation left from the timing contracts ifdef

Reply:
also, and i know this is really picky :), could we move this above the first `GUARD` definition. I tried to keep things grouped so it's easier to maintain

Reply:
No problem, maintainability is important :) I updated as you suggested.